packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -2,20 +2,170 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
+
 import 'basic_types.dart';
+import 'diagnostics.dart';
 import 'print.dart';
 
+// Examples can assume:
+// dynamic element;
+// dynamic color;
+// dynamic _ErrorDiagnostic;
+
 /// Signature for [FlutterError.onError] handler.
 typedef FlutterExceptionHandler = void Function(FlutterErrorDetails details);
 
 /// Signature for [FlutterErrorDetails.informationCollector] callback
-/// and other callbacks that collect information into a string buffer.
-typedef InformationCollector = void Function(StringBuffer information);
+/// and other callbacks that collect information describing an error.
+typedef InformationCollector = Iterable<DiagnosticsNode> Function();
+
+class _ErrorDiagnostic extends DiagnosticsProperty<List<Object>> {
+  /// In debug builds, a kernel transformer rewrites calls to the default
+  /// constructors for [ErrorSummary], [ErrorDetails], and [ErrorHint] to use
+  /// this constructor.
+  ///
+  /// {@tool sample}
+  /// ```dart
+  /// _ErrorDiagnostic('Element $element must be $color')
```

Comment:
I don't see this _ErrorDiagnostic constructor defined anywhere? Is this supposed to be a placeholder for ErrorSummary, ErrorDetails or ErrorHint?

Reply:
Might be worthwhile to point that out or use a concrete example.

Reply:
I went ahead and added a default constructor for this class.